src/Interpreters/Streaming/Aggregator.cpp

```diff
@@ -142,6 +143,39 @@ void AggregatedDataVariants::convertToTwoLevel()
     }
 }
 
+void AggregatedDataVariants::updateMetrics(AggregatedDataMetrics & metrics) const
```

Comment:
How expensive is for this call ?

Reply:
I have log it, looks good.  `0 ms`.
```c++
LOG_INFO(
        log,
        "Took {} milliseconds to log metrics. Aggregated data metrics: {}",
        end - start,
        aggregated_data_metrics.string());
```

Reply:
let's track microseconds level probably.

Reply:
> let's track microseconds level probably.

make sense, I can modify it in another pr